torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -483,6 +483,37 @@ void GraphExecutorImplBase::run(Stack& stack) {
   last_executed_optimized_graph = plan.graph;
 }
 
+c10::intrusive_ptr<Future> GraphExecutorImplBase::runAsync(Stack& stack) {
+  TORCH_CHECK(
+      stack.size() >= num_inputs,
+      "expected ",
+      num_inputs,
+      " inputs, but got only ",
+      stack.size());
+
+  C10_LOG_API_USAGE_ONCE("torch.graph_executor.runAsync");
+  logging::getLogger()->addStatValue(
+      logging::runtime_counters::GRAPH_EXECUTOR_INVOCATIONS, 1.0);
+
+  struct Frame {
+    explicit Frame(ExecutionPlan eplan)
+      : plan(std::move(eplan)),
+        state(plan.code) {
+    }
+    ExecutionPlan plan;
+    InterpreterState state;
+  };
+  auto frame = std::make_shared<Frame>(
+    getPlanFor(stack, GraphExecutor::getDefaultNumBailOuts()));
```

Comment:
Is this how we keep the stack alive?

Reply:
No, this part does not keep the Stack alive, but rather the Code. It's not 100% clear that this was necessary, actually (and actually it would probably be ok not to heap allocate it, but rather stack alloc it and std::move to a heap allocated struct, since the internals are all pimpl-to-shared-ptr, but this felt aggressive)